databuilder/databuilder/extractor/es_base_extractor.py

```diff
@@ -47,13 +74,48 @@ def _get_indexes(self) -> Dict:
     def _get_index_mapping_properties(self, index: Dict) -> Optional[Dict]:
         mappings = index.get('mappings', dict())
 
+        # Mapping types were removed in Elasticsearch 7. As a result, index mappings are formatted differently.
+        # See https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html
+        no_mapping_types = False
+        version = self._get_es_version()
+        if len(version) > 0:
+            version_numbers = version.split('.')
+            if int(version_numbers[0]) >= 7:
+                no_mapping_types = True
+
         try:
-            properties = list(mappings.values())[0].get('properties', dict())
+            if no_mapping_types:
+                properties = mappings.get('properties', dict())
+            else:
+                properties = list(mappings.values())[0].get('properties', dict())
         except Exception:
             properties = dict()
 
         return properties
 
+    def _get_nested_columns(self,
```

Comment:
please rename to _get_attributes, because:
1. there are no columns in elasticsearch, only attributes/fields of a document
2. nested type is sth else (see my previous comments) https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html#nested

Reply:
Agreed. Updated.